src/spanningtrees/kruskal.jl

```diff
@@ -1,5 +1,5 @@
 struct KruskalHeapEntry{T<:Real}
-    edge::Edge
+    edge::AbstractEdge
```

Comment:
I think we want to do this differently, since `AbstractEdge` is, well, abstract. How about
```
struct KruskalHeapEntry{T<:Real, U<:AbstractEdge)
    edge::U
    dist::T
end
```
?

Reply:
Agreed. Whatever the graph type, the edges on a graph should be of the same type.